ui/src/routes/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/create-run/-create-run-utils.ts

```diff
@@ -592,12 +622,85 @@ export function formValuesToPayload(
       }
     : undefined;
 
+  //
+  // Reporter configuration
+  //
+
+  // Check if CycloneDX, SPDX, and/or NOTICE file reports are enabled in the form,
+  // and configure them to use all output formats, accordingly.
+
+  // Unfortunately, OpenAPI's PluginConfiguration always requires 'secrets' object,
+  // and it cannot be "undefined", so we need to pass in an empty secrets object instead,
+  // which produces an eslint error about empty object types allowing non-nullish values.
+
+  /* eslint-disable @typescript-eslint/no-empty-object-type */
+
+  const cycloneDxEnabled =
+    values.jobConfigs.reporter.formats.includes('CycloneDX');
+  const spdxDocumentEnabled =
+    values.jobConfigs.reporter.formats.includes('SpdxDocument');
+  const noticeFileEnabled =
+    values.jobConfigs.reporter.formats.includes('PlainTextTemplate');
```

Comment:
Hard-coding this not not really scale well if we had additional formats. Also see my comment below.